### ISSUE TITLE
 node-red-node-mysql Uncaught Exception after update to 0.1.3

### ISSUE THREAD

**Issue by Ashfaak** [node-red/node-red-nodes#771]
- [ ] Node-RED version: 1.2.7
- [ ] node.js version: 12.21.0
- [ ] npm version: 6.14.11
- [ ] Platform/OS: Debian 10
- [ ] Browser:

### Which node are you reporting an issue on?
node-red-node-mysql

### What are the steps to reproduce?
Update node-red-node-mysql from 0.1.1 or 0.1.2 to 0.1.3
Node-RED runs as expected
6-7 minutes and a few hundred queries later
(no errors caught by catch node)

### What happens?
Node-RED suffers Uncaught Exception and dies or restarts depending on how it was started.

Log as follows:
```node-red-start

Start Node-RED

Once Node-RED has started, point a browser at http://xxx:1880
On Pi Node-RED works better with the Firefox or Chrome browser

Use   sudo systemctl enable nodered.service  to autostart Node-RED at every boot
Use   sudo systemctl disable nodered.service to disable autostart on boot

To find more nodes and example flows - go to http://flows.nodered.org

Starting as a normal user.
14 Mar 15:23:42 - [info]

Welcome to Node-RED
===================

14 Mar 15:23:42 - [info] Node-RED version: v1.2.7
14 Mar 15:23:42 - [info] Node.js  version: v12.21.0
14 Mar 15:23:42 - [info] Linux 4.19.0-11-cloud-amd64 x64 LE
14 Mar 15:23:42 - [info] Loading palette nodes
14 Mar 15:23:45 - [info] Dashboard version 2.28.1 started at /
14 Mar 15:23:46 - [info] Settings file  : /home/Ashfaak/.node-red/settings.js
14 Mar 15:23:46 - [info] Context store  : 'memory' [module=memory]
14 Mar 15:23:46 - [info] Context store  : 'file' [module=localfilesystem]
14 Mar 15:23:46 - [info] User directory : /home/Ashfaak/.node-red
14 Mar 15:23:46 - [warn] Projects disabled : editorTheme.projects.enabled=false
14 Mar 15:23:46 - [info] Flows file     : /home/Ashfaak/.node-red/flows_servername.json
14 Mar 15:23:46 - [info] Server now running at https://127.0.0.1:1880/admin/
14 Mar 15:23:46 - [warn]

---------------------------------------------------------------------
Your flow credentials file is encrypted using a system-generated key.

If the system-generated key is lost for any reason, your credentials
file will not be recoverable, you will have to delete it and re-enter
your credentials.

You should set your own key using the 'credentialSecret' option in
your settings file. Node-RED will then re-encrypt your credentials
file using your chosen key the next time you deploy a change.
---------------------------------------------------------------------

14 Mar 15:23:46 - [info] Starting flows
14 Mar 15:23:47 - [info] Started flows
14 Mar 15:23:47 - [error] [function:Daily Ave meters, Error calc -> Chart] TypeError: Cannot read property 'tagPara  meters' of undefined
14 Mar 15:23:47 - [error] [function:Hourly Ave meters, Error calc -> Chart] TypeError: Cannot read property 'tagPar  ameters' of undefined
14 Mar 15:23:47 - [error] [function:Periodic Ave meters, Error calc -> Chart] TypeError: Cannot read property 'tagP  arameters' of undefined
14 Mar 15:23:47 - [error] [function:ProdMonth Ave meters, Error calc -> Chart] TypeError: Cannot read property 'tag  Parameters' of undefined
14 Mar 15:23:47 - [error] [function:PeriodicMonth Ave meters, Error calc -> Chart] TypeError: Cannot read property   'tagParameters' of undefined
14 Mar 15:23:47 - [error] [function:Periodic Ave meters, Error calc, HFO/Coal calc] TypeError: Cannot read property   'tagParameters' of undefined
14 Mar 15:23:47 - [error] [function:Daily Ave meters, Error calc, HFO/Coal calc] TypeError: Cannot read property 't  agParameters' of undefined
14 Mar 15:23:47 - [error] [function:Hourly Ave meters, Error calc, HFO/Coal calc] TypeError: Cannot read property '  tagParameters' of undefined
14 Mar 15:23:47 - [error] [function:ProdMonth Ave meters, Error calc, HFO/Coal calc] TypeError: Cannot read propert  y 'tagParameters' of undefined
14 Mar 15:23:47 - [error] [function:PeriodicMonth Ave meters, Error calc -> Chart] TypeError: Cannot read property   'tagParameters' of undefined
14 Mar 15:23:47 - [error] [function:Daily Ave meters, Error calc -> Chart] TypeError: Cannot read property 'tagPara  meters' of undefined
14 Mar 15:23:47 - [error] [function:Hourly Ave meters, Error calc -> Chart] TypeError: Cannot read property 'tagPar  ameters' of undefined
14 Mar 15:23:47 - [error] [function:Periodic Ave meters, Error calc -> Chart] TypeError: Cannot read property 'tagP  arameters' of undefined
14 Mar 15:23:47 - [error] [function:ProdMonth Ave meters, Error calc -> Chart] TypeError: Cannot read property 'tag  Parameters' of undefined
14 Mar 15:23:47 - [error] [function:Periodic Ave meters, Error calc -> Chart] TypeError: Cannot read property 'tagP  arameters' of undefined
14 Mar 15:23:53 - [error] [function:latest info update] TypeError: Cannot read property 'originalValue' of undefined
14 Mar 15:23:53 - [error] [function:latest info update] TypeError: Cannot read property 'originalValue' of undefined
14 Mar 15:23:53 - [error] [function:latest info update] TypeError: Cannot read property 'originalValue' of undefined
14 Mar 15:23:57 - [error] [function:latest info update] TypeError: Cannot read property 'originalValue' of undefined
14 Mar 15:23:57 - [error] [function:latest info update] TypeError: Cannot read property 'originalValue' of undefined
14 Mar 15:23:57 - [error] [function:latest info update] TypeError: Cannot read property 'originalValue' of undefined
14 Mar 15:24:28 - [warn] Communication send error: Error: WebSocket is not open: readyState 2 (CLOSING)
14 Mar 15:24:42 - [error] [function:latest info update] TypeError: Cannot read property 'originalValue' of undefined
14 Mar 15:24:54 - [error] [function:Periodic Ave meters, Error calc -> Chart] TypeError: Cannot read property 'originalValues' of undefined
14 Mar 15:25:02 - [error] [function:latest info update] TypeError: Cannot read property 'originalValue' of undefined
14 Mar 15:25:11 - [error] [function:Periodic Ave meters, Error calc -> Chart] TypeError: Cannot read property 'originalValues' of undefined
14 Mar 15:25:28 - [error] [function:latest info update] TypeError: Cannot read property 'originalValue' of undefined
14 Mar 15:26:11 - [info] Stopping modified nodes
14 Mar 15:26:11 - [info] Stopped modified nodes
14 Mar 15:26:11 - [info] Starting modified nodes
14 Mar 15:26:11 - [info] Started modified nodes
14 Mar 15:26:16 - [error] [function:Periodic Ave meters, Error calc -> Chart] TypeError: Cannot read property 'originalValues' of undefined
14 Mar 15:30:53 - [red] Uncaught Exception:
14 Mar 15:30:53 - Error: Connection already released
    at Pool.releaseConnection (/home/Ashfaak/.node-red/node_modules/mysql/lib/Pool.js:138:13)
    at PoolConnection.release (/home/Ashfaak/.node-red/node_modules/mysql/lib/PoolConnection.js:35:15)
    at Query.<anonymous> (/home/Ashfaak/.node-red/node_modules/mysql/lib/Pool.js:211:12)
    at Object.onceWrapper (events.js:420:28)
    at Query.emit (events.js:326:22)
    at Query.Sequence.end (/home/Ashfaak/.node-red/node_modules/mysql/lib/protocol/sequences/Sequence.js:86:12)
    at Query._handleFinalResultPacket (/home/Ashfaak/.node-red/node_modules/mysql/lib/protocol/sequences/Query.js:149:8)
    at Query.EofPacket (/home/Ashfaak/.node-red/node_modules/mysql/lib/protocol/sequences/Query.js:133:8)
    at Protocol._parsePacket (/home/Ashfaak/.node-red/node_modules/mysql/lib/protocol/Protocol.js:291:23)
    at Parser._parsePacket (/home/Ashfaak/.node-red/node_modules/mysql/lib/protocol/Parser.js:433:10)
````

Please ignore the TypeErrors - I've been meaning to fix them.

### What do you expect to happen?
Error to be catchable so that NR does not restart, or error to be handled somehow.

### Please tell us about your environment:
This Node-RED instance is used as a dashboard hosted on GCP. It queries data from a Mysql DB hosted by another party.
I have multiple mysql nodes on a single flow accessing the same mysql server.

### Other notes
This error does not occur on ver 0.1.1 or 0.1.2. I've downgraded to 0.1.2 for now.
This instance of NR restarted multiple times because of this issue - i.e. it happened consistently.

**Comment by dceejay** [comment]
Hi - thanks for reporting - I have undone another change that I thought would improve pool handling (but apparently not) - pushed as version 0.1.4 - can you retry with that ? Thanks

**Comment by Ashfaak** [comment]
Hi @dceejay 

Using 0.1.3 I tried changing the flow by removing 80% of the mysql nodes, but to no avail.

Tested 0.1.4 and it has been working without errors for a just over 2 hours.

Tried reading a bit of the mysql.js documentation to see if I could be of any help. Most of it is a bit too advanced for me right now. I found this:

`pool.query()` is shortcut for `pool.getConnection() + connection.query() + connection.release()`: https://github.com/mysqljs/mysql/issues/1202 

Pros / cons of using` pool.getConnection()` over `pool.query()`: https://github.com/mysqljs/mysql/issues/1668

Issue with the same error: https://github.com/mysqljs/mysql/issues/2048

It seems to me that within the context of a NR node it makes sense to use `pool.query()`.

Thank you
